tests/Integration/Behaviour/Features/Scenario/CartRule/set_cart_rule_restrictions.feature

```diff
@@ -0,0 +1,96 @@
+      | quantity_per_user                      | 12                     |
+      | free_shipping                          | false                  |
+      | code                                   | rule_50_percent        |
+      | reduction_percentage                   | 50                     |
+      | reduction_apply_to_discounted_products | false                  |
+      | discount_application_type              | order_without_shipping |
+    And I create cart rule "rule_70_percent" with following properties:
+      | name[en-US]                            | Half the price         |
+      | is_active                              | true                   |
+      | allow_partial_use                      | true                   |
+      | priority                               | 3                      |
+      | valid_from                             | 2022-01-01 11:00:00    |
+      | valid_to                               | 3001-01-01 12:00:00    |
+      | total_quantity                         | 10                     |
+      | quantity_per_user                      | 12                     |
+      | free_shipping                          | false                  |
+      | code                                   | rule_70_percent        |
+      | reduction_percentage                   | 70                     |
+      | reduction_apply_to_discounted_products | false                  |
+      | discount_application_type              | order_without_shipping |
+    And cart rule "rule_free_shipping_1" should have the following properties:
+      | restricted cart rules |  |
+    And cart rule "rule_50_percent" should have the following properties:
+      | restricted cart rules |  |
+    And cart rule "rule_70_percent" should have the following properties:
+      | restricted cart rules |  |
+
+  Scenario: Restrict cart rule combinations
+    When I restrict following cart rules for cart rule rule_free_shipping_1:
+      | rule_50_percent |
+    Then cart rule "rule_free_shipping_1" should have the following properties:
+      | restricted cart rules | rule_50_percent |
+    And cart rule "rule_50_percent" should have the following properties:
+      | restricted cart rules | rule_free_shipping_1 |
+    And cart rule "rule_70_percent" should have the following properties:
+      | restricted cart rules |  |
+    When I restrict following cart rules for cart rule rule_free_shipping_1:
+      | rule_50_percent |
+      | rule_70_percent |
+    Then cart rule "rule_free_shipping_1" should have the following properties:
+      | restricted cart rules | rule_50_percent,rule_70_percent |
+    And cart rule "rule_50_percent" should have the following properties:
+      | restricted cart rules | rule_free_shipping_1 |
+    And cart rule "rule_70_percent" should have the following properties:
+      | restricted cart rules | rule_free_shipping_1 |
+
+  @restore-cart-rules-before-scenario
```

Comment:
It's really not normal that you can't use this on the top of the file That's quite worrying because we rely on this behaviour with other tags and other behat feature files

Reply:
I think it is, because it is beforescenario so its used on scenarios (and not whole feature)